aptos/token_bridge/sources/bridge_state.move

```diff
@@ -193,46 +274,68 @@ module token_bridge::bridge_state {
             origin_address = token_hash::get_bytes(&token_address);
         };
 
-        // TODO - normalize amount by using helpers from utils.move
-        let normalized_amount = u256::from_u64(amount);
-        // TODO - normalize relayer fee
-        let normalized_relayer_fee = u256::from_u128(relayer_fee);
-        let wormhole_fee = u256::from_u64(0);
+        let decimals_token = coin::decimals<CoinType>();
+        let decimals_aptos = coin::decimals<AptosCoin>();
+
+        let normalized_amount = utils::normalize_amount(u256::from_u64(amount), decimals_token);
+        let normalized_relayer_fee = utils::normalize_amount(u256::from_u64(relayer_fee), decimals_aptos);
 
         let transfer_result: TransferResult = transfer_result::create(
             origin_chain,
             origin_address,
             normalized_amount,
             normalized_relayer_fee,
-            wormhole_fee
+            u256::from_u64(wormhole_fee)
         );
         transfer_result
     }
 
-    public entry fun transfer_tokens_with_signer<CoinType>(sender: &signer, amount: u64, relayer_fee: u128): TransferResult acquires State, OriginInfo {
-        let coins = coin::withdraw<CoinType>(sender, amount);
-        transfer_tokens<CoinType>(coins, relayer_fee)
-    }
-
     public(friend) fun log_transfer(
         token_chain: U16,
         token_address: vector<u8>,
         amount: U256,
         recipient_chain: U16,
         recipient: vector<u8>,
         fee: U256,
-        message_fee: Coin<AptosCoin>,
+        wormhole_fee: Coin<AptosCoin>,
         nonce: u64
     ): u64 acquires State{
-        let fee_value = coin::value<AptosCoin>(&message_fee);
-        assert!(u256::compare(&u256::from_u64(fee_value), &amount)==1, E_FEE_EXCEEDS_AMOUNT);
-        // TODO - payloadID is 1 for token transfer?
+        // TODO - check fee values
+        //let fee_value = coin::value<AptosCoin>(&wormhole_fee);
+        //assert!(u256::compare(&u256::from_u64(fee_value), &amount)==1, E_FEE_EXCEEDS_AMOUNT);
         let transfer = transfer::create(1, amount, token_address, token_chain, recipient, recipient_chain, fee);
         let payload = transfer::encode(transfer);
         publish_message(
             nonce,
             payload,
-            message_fee,
+            wormhole_fee,
+        )
+    }
+
+    public(friend) fun log_transfer_with_payload(
+        token_chain: U16,
+        token_address: vector<u8>,
+        amount: U256,
+        recipient_chain: U16,
+        recipient: vector<u8>,
+        wormhole_fee: Coin<AptosCoin>,
+        nonce: u64,
+        payload: vector<u8>
+    ): u64 acquires State{
+        let transfer = transfer_with_payload::create(
+            amount,
+            token_address,
+            token_chain,
+            recipient,
+            recipient_chain,
+            to_bytes<address>(&@token_bridge), //TODO - is token bridge the only one who will ever call log_transfer_with_payload?
```

Comment:
no, we'll have to roll something similar to `EmitterCapability`, but at the token bridge level. for now I reused `EmitterCapability` in `complete_transfer` but we should have a token bridge version of that stuff really.

Reply:
we'll probably have to discuss this later, I'm not sure what that implementation would look like